blocks/library/image/image-size.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+
+class ImageSize extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			width: undefined,
+			height: undefined,
+		};
+		this.bindContainer = this.bindContainer.bind( this );
+	}
+
+	bindContainer( ref ) {
+		this.container = ref;
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.src !== prevProps.src ) {
+			this.setState( {
+				width: undefined,
+				height: undefined,
+			} );
+			this.fetchImageSize();
+		}
+	}
+
+	componentDidMount() {
+		this.fetchImageSize();
+	}
+
+	componentWillUnmout() {
+		if ( this.image ) {
+			this.image.src = noop;
+		}
+	}
+
+	fetchImageSize() {
+		this.image = new window.Image();
+		this.image.onload = () => {
+			const maxWidth = this.container.clientWidth;
+			const ratio = this.image.height / this.image.width;
+			const width = this.image.width < maxWidth ? this.image.width : maxWidth;
+			const height = this.image.width < maxWidth ? this.image.height : maxWidth * ratio;
+			this.setState( { width, height } );
```

Comment:
I did some testing. It seems to prevent `onload` callback if changed while in-flight. Though I think we'd ought to set to an empty string, not sure what the expected behavior of a function is.

![image](https://user-images.githubusercontent.com/1779930/28975251-6067ea02-7907-11e7-92a2-5294b27bb83e.png)